### ISSUE TITLE
No HTML

### ISSUE THREAD

**Issue by dudde0man** [tectonicus/tectonicus#240]
When i try to create a map the program runs to completion, but doesn't provide a HTML file. All the PNGs seem to generate. I've tried to troubleshoot but there doesn't seem to be any errors, and i have no clue if i'm just missing something.
I would be eternally grateful if anyone would help <3.

[tectonicus.log](https://github.com/tectonicus/tectonicus/files/15210488/tectonicus.log)
[config.txt](https://github.com/tectonicus/tectonicus/files/15210501/New.Text.Document.txt)

**Comment by skoeven** [comment]
From reading your log file it seems that the tile rendering is not actually finishing. If it had finished you would see "Base tile render complete" in the log followed by a bunch log lines for downsampling and writing out javascript and html files.  Unfortunately I'm not sure what's happening or how it could get interrupted like that with no errors.  I would have you turn verbose logging on but there is a bug in the current version so that verbose mode doesn't work.  I will be releasing a new Tectonicus version in the next day or so that will include a fix for this bug. With that version you can turn on verbose logging and we can get some more information on where the render is stopping.

**Comment by skoeven** [comment]
It took longer than planned, but there is a new version of Tectonicus out that fixes the verbose logging: https://github.com/tectonicus/tectonicus/releases/tag/v2.30.1

You can run turn on verbose logging by adding the -v command line option:
`java -jar .\Tectonicus-2.30.1.jar -c config.xml -v`

**Comment by dudde0man** [comment]
Thanks for the update!
The update made the process provide an error message so i managed to fix the issue.
It was a memory issue with "GCLocker too often allocating x words - java out of memory java heap space". I fixed it by allocating 6g's of ram specifically on startup, it didn't even let me run the program with 4g's allocated. If you want the logs for something i can post them if needed.
I also quickly want to say that i have used quite a few mapping tools over the years, and i'm highly impressed with your program, you've done an amazing job, thanks for your work and help.

**Comment by skoeven** [comment]
Great! I'm glad you were able to resolve the problem.  If you get a chance could you post your logs? Especially if you have the logs with the error message. I'm curious to see where it was happening.

**Comment by dudde0man** [comment]
Here are the logs. I can't find any error messages in the log so i included a picture of the error from the cmd. I've run the process like this multiple times and the issue seems to occur around 2600 tiles every time.
[tectonicus.log]
(https://github.com/tectonicus/tectonicus/files/15401726/tectonicus.log)
![image](https://github.com/tectonicus/tectonicus/assets/44510308/515d0850-5d4e-490a-a5af-e8e13e6db07e)

**Comment by dudde0man** [comment]
[tectonicus.log](https://github.com/tectonicus/tectonicus/files/15401892/tectonicus.log)

**Comment by skoeven** [comment]
Thank you, that's very helpful!
If you don't mind I have a couple of other questions:
1. Are you using the java -Xms -Xmx properties to set the amount of heap memory to get it to work?
2. How much memory does your machine have?
3. Could you screenshot the output of this command: java -XX:+PrintFlagsFinal -version | findstr HeapSize

2600 tiles is not very much so I want to try and narrow down what the problem is so we can either possibly decrease our memory usage or at least update the docs to let people know they may need to manually provide more memory to the JVM.

**Comment by dudde0man** [comment]
1. i used -Xmx6g. I didn't add or change anything else.
2. I have 8 ram
3. 
![image](https://github.com/tectonicus/tectonicus/assets/44510308/07afa48d-2f36-4d99-8810-dc431fc28002)

**Comment by dudde0man** [comment]
I'm not sure if i should open a new issue or not by I've ran into another problem. When i generate maps the stone turns white. There doesn't seem to be any errors in the log and other textures seem fine. This issue persists in views.
![image](https://github.com/tectonicus/tectonicus/assets/44510308/af5b57e4-3f46-45de-86be-6716802ddc63)
The issue was most likely present in all previous logs but i'll post this to make sure.
[tectonicus.log](https://github.com/tectonicus/tectonicus/files/15433328/tectonicus.log)

**Comment by skoeven** [comment]
First, thanks for the memory info you provided. I've updated the readme to let people know how to increase the memory available to Java. Also Tectonicus now logs out the current max heap memory size for convenience.
<img width="266" alt="memory" src="https://github.com/tectonicus/tectonicus/assets/466126/90b0e519-f3c6-49d8-b220-58652b1bddae">

Second, it looks like you've discovered a lighting issue with 1.20.  I had seen it myself before but it didn't really register until you mentioned it. I went and did a comparison:
<img width="527" alt="lightingChange" src="https://github.com/tectonicus/tectonicus/assets/466126/43e55033-ac9a-480e-aa30-984a5464d0b2">

I've created a new issue because this will unfortunately require some investigation. I'll go ahead and close this issue.
